DataManagementSystem/DB/FileCatalogComponents/Utilities.py

```diff
@@ -60,9 +57,52 @@ def checkArgumentDict( path ):
       uList = mUrl.split('/')
       if len( uList ) >= 2 and uList[1] == 'grid':  
         mUrl = mUrl[5:] 
-    normpath = os.path.normpath( mUrl )
-    urls[normpath] = pathDict[url]
-  return S_OK( urls )
+    normPath = os.path.normpath( mUrl )
+    urls[normPath] = pathDict[url]
+    if normPath != url:
+      urlMap[normPath] = url
+  if generateMap:
+    return S_OK( ( urls, urlMap ) )
+  else:
+    return S_OK( urls )
+
+def checkCatalogArguments( f ):
+  """ Decorator to check arguments of FileCatalog calls in the clients
+  """
+  def processWithCheckingArguments(*args, **kwargs):
+
+    checkFlag = kwargs.pop( 'NoLFNChecking', True )
```

Comment:
so if 'NoLFNChecking == True', we do the check ? Shouldn't it be the other way around ?